### DISCUSSION TITLE
No module named 'pydantic_core._pydantic_core'

### DISCUSSION THREAD

**Discussion by rhorn17** [pydantic/pydantic#6935] [Question] [answered]
Hi, I'm running into this error on AWS when trying to call a model through the replicate API.  Replicate has pydantic as a dependency and while it works great locally (in Pycharm on a Mac, arm64), the exact same code fails with this error on AWS (lambda function running in arm64 architecture, utilizing an arm64 layer with the replicate libraries, including pydantic).  Any suggestions would be greatly appreciated.  I tried dropping down to version 2.0 of pydantic, but that didn't help.  Thanks!

**Comment by rhorn17** [comment, marked as answer]
Never mind, I figured it out.  I had to go get the "pydantic_core-2.4.0-cp310-cp310-manylinux_2_17_aarch64.manylinux2014_aarch64.whl" from the pydantic_core website, unpack that (can't pip install that on a Mac), build a layer with that version, and upload that to AWS.  Then is worked.

**Reply by toddmetheny** [reply]
Trying to figure out how to do this. I downloaded the zip. Did you create the layer in the UI and then upload the zip file?

**Reply by rhorn17** [reply]
Sorry for the delay in responding.  So I downloaded the .whl file and expanded it locally.  Then I moved the included packages/folders under a folder named "python" (important it's called 'python') with my other packages.  I zipped that folder up and uploaded it as a layer in AWS.  Does this help?

**Reply by toddmetheny** [reply]
It is helpful and I appreciate you responding. 

I just tried it again and did the following -- downloaded the `whl` file from the pydantic core github releases page. Ran unzip in the terminal. This produced two folders. I moved them into a folder called python. Ran zip on that file. Navigated to the lambda UI and created a new layer called Pydantic, uploaded the python.zip, selected `x86_64` and selected the appropriate python version. Then went back to the lambda in the UI and added the customer layer. Waited for the lambda to tell me it had updated. However, when I click test in the lambda UI I still get this response:

```
{
  "errorMessage": "Unable to import module 'index': No module named 'pydantic_core._pydantic_core'",
  "errorType": "Runtime.ImportModuleError",
  "stackTrace": []
}
```

Download was from here: https://github.com/pydantic/pydantic-core/releases 

File structure:
![Screenshot 2024-12-12 at 10 37 30 AM](https://github.com/user-attachments/assets/1d102afd-94cc-46aa-ac1a-5ad558342bca)

Any thoughts on what I'm doing wrong?

**Reply by rhorn17** [reply]
if you downloaded the same .whl file as I did, you will need to select 'arm64' for the architecture for the lambda as that specific .whl file is specific to that chipset.  If you really do want x86_64, you'll need to download a different .whl file.  Not sure of the exact name you'll need, but it will not contain the 'aarch64' tag in the name.  Hope this helps

**Reply by rhorn17** [reply]
Just seeing your .whl file in your screen print now sorry...not sure what platform that is, but just know pedantic will give you that exact error if you don't have the correct 1) platform (manylinux for AWS lambda), and 2) chipset (whichever you choose has to be consistent with your other packages and with the architecture you select for the lambda).  Also, you have to make sure you have the correct version and build of python (cp310 is the 'C' version of python v. 3.10)

**Reply by toddmetheny** [reply]
@rhorn17 Took me a few tries but finally got the layer to work! I had to install supabase in a docker image with the correct architecture flagged. Thank you for your responses and help putting me on the right track.

**Reply by rhorn17** [reply]
Absolutely!  Glad I could help a little, and glad you were able to get it to work!

**Comment by ramunecroft** [comment]
Is this going to be fixed someday?  . . .

**Comment by danielWagnerr** [comment]
This bug isn't fixed yet?

**Reply by maksimp13** [reply]
the same here, very annoying...

**Reply by danielWagnerr** [reply]
I "fixed" this bug by running the pip install on a Ubuntu  x86_64 machine with Python 3.11 installed, if it helps...

**Reply by Surya-Regalla** [reply]
what command did u exactly used and on which terminal??

**Reply by rhorn17** [reply]
I was using a Mac terminal window (zshell), and just unzipped the .whl file using "unzip"

**Reply by danielmarzan** [reply]
With serverless it would be:
frameworkVersion: "3"
plugins:
  - serverless-python-requirements
provider:
  name: aws
  runtime: python3.11
  **architecture: x86_64** //arm64 does not work with pedantic_core:latest at the moment
pythonRequirements:
    dockerizePip: true 
    dockerRunCmdExtraArgs: [ '--platform', 'linux' ]
    dockerImage: public.ecr.aws/sam/build-python3.11:latest
    useStaticCache: true
    useDownloadCache: true
    slim: true
    slimPatternsAppendDefaults: false
    slimPatterns:
      - "**/*.py[c|o]"
      - "**/__pycache__*"
    layer:
      name: ${self:service}-layer-deployment-${self:custom.stage}
      description: Python requirements lambda layer
      compatibleRuntimes:
        - python3.11

**Comment by alexlatif** [comment]
crazy this is still not fixed

**Reply by alexlatif** [reply]
for anyone using "PDM" ensure you target your platforms of your lock file.
